tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -729,7 +735,17 @@ public void reset() {
 
     @Override
     public Set<TaskUrl> getTaskUrls() {
-      if (session != null && session.allTasksScheduled()) {
+      LOG.info("Client requesting TaskUrls!");
+      if (singleNode && proxyUrl != null) {
+        return new HashSet<TaskUrl>(){{
```

Comment:
I suggest avoiding double-brace initialization. It's generally not a good Java practice: https://stackoverflow.com/a/27521360/1128392

Reply:
Good to know, thanks!